src/react/databrowser/objects/__tests__/Metadata.test.jsx

```diff
@@ -0,0 +1,102 @@
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        expect(component.find('input[name="mdKeyType"]').prop('value')).toBe('');
+        expect(component.find('input#metadata-input-value').prop('value')).toBe('');
+        expect(component.find('SelectBox')).toHaveLength(1);
+    });
+
+    it('should add new key/value metadata and should trigger function if save button is pressed', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        // open dropdown menu
+        component.find('input').at(0).simulate('change');
+
+        // check all options are present
+        const optionsComponents = component.find('Option');
+        optionsComponents.forEach((optionComponent, index) => {
+            expect(optionComponent.find('div.sc-select-option-label').text()).toBe(optionLabels[index]);
+        });
+
+        // select a key
+        optionsComponents.at(0).simulate('click');
+
+        // check dropdown menu disappear
+        expect(component.find('Option')).toHaveLength(0);
+
+        // add a value to the key
+        component.find('input#metadata-input-value').simulate('change', { target: { value: 'value1' } });
+
+        // click on add button
+        component.find(AddButton).simulate('click');
+
+        // check new key/value added
+        expect(component.find('SelectBox')).toHaveLength(2);
+
+        // check if function when pressing save button is triggered
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(0);
+        component.find('Button#metadata-button-save').simulate('click');
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('remove button and add button should be disabled', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        expect(component.find(AddButton).children().prop('disabled')).toBe(true);
+        expect(component.find(SubButton).children().prop('disabled')).toBe(true);
+    });
+
+    it('should render SelectBox with key/value pass in props', () => {
+        const { component } = reduxMount(<Metadata
+            objectMetadata={{ ...OBJECT_METADATA, metadata: [{ key: 'cache-control', value: 'value1' }] }}/>);
+
+        expect(component.find('input[name="mdKeyType"]').prop('value')).toBe('cache-control');
+        expect(component.find('input#metadata-input-value').prop('value')).toBe('value1');
+        expect(component.find('SelectBox')).toHaveLength(1);
+    });
+
+    it('should delete key/value if remove button is pressed', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={{
+            ...OBJECT_METADATA,
+            metadata: [{ key: 'cache-control', value: 'value1' }, { key: 'content-encoding', value: 'value2' }],
+        }}/>);
+
+        // check if everything render well
+        expect(component.find('input[name="mdKeyType"]').at(0).prop('value')).toBe('cache-control');
```

Comment:
We can first check that two "rows" are returned before checking their values.

Reply:
done